main.go

```diff
@@ -107,6 +108,8 @@ type metricsConfig struct {
 	writeEndpoint *url.URL
 	tenantHeader  string
 	tenantLabel   string
+	// enable metrics if at least one {read|write}Endpoint} is provided.
+	enabled bool
```

Comment:
I think we need to exit with an error of neither logging nor metrics are enabled. If the administrator doesn't specify any endpoints then the API can't do anything. IMO it's the logical extrapolation of the current logic where we said "at least a metrics endpoint is mandatory". Now we would be saying "specifying at least a logging or a metrics endpoint is mandatory"